android/tangram/jni/platform_android.cpp

```diff
@@ -245,20 +256,25 @@ unsigned char* bytesFromFileSystem(const char* _path, unsigned int* _size) {
 
 }
 
-std::string stringFromFile(const char* _path, PathType _type) {
+std::string stringFromFile(const char* _path, PathType _type, const char* _resourceRoot) {
 
     unsigned int length = 0;
-    unsigned char* bytes = bytesFromFile(_path, _type, &length);
+    unsigned char* bytes = bytesFromFile(_path, _type, &length, _resourceRoot);
     std::string out(reinterpret_cast<char*>(bytes), length);
     free(bytes);
 
     return out;
 
 }
 
-unsigned char* bytesFromFile(const char* _path, PathType _type, unsigned int* _size) {
+unsigned char* bytesFromFile(const char* _path, PathType _type, unsigned int* _size, const char* _resourceRoot) {
```

Comment:
Discussing more on this:

Currently file loading is required in 3 different places!
- sceneLoader: to load textures from resource Paths (Scene can handle this).
- fontContext: to load atlas (a const ref to resource root from Scene's resourceRoot member can be set to handle this internally in fontContext).
- DataSource: (could be internal/absolute or relative to resource path). Scene can not handle this. One proposal could be make DataSource aware of the same `resourceRoot` which Scene has and handle it internally in DataSource. But I am not sure about it.